app/org/sagebionetworks/bridge/dynamodb/DynamoSchedulePlanDao.java

```diff
@@ -88,39 +95,100 @@ public SchedulePlan getSchedulePlan(StudyIdentifier studyIdentifier, String guid
+        checkNotNull(plan);
         
         plan.setStudyKey(studyIdentifier.getIdentifier());
         plan.setModifiedOn(DateUtils.getCurrentMillisFromEpoch());
+        
+        forEachCriteria(plan, scheduleCriteria -> persistCriteria(scheduleCriteria));
         mapper.save(plan);
         return plan;
     }
 
     @Override
     public void deleteSchedulePlan(StudyIdentifier studyIdentifier, String guid) {
-        checkNotNull(studyIdentifier, "Study identifier is null");
-        checkArgument(StringUtils.isNotBlank(guid), "Plan GUID is blank or null");
+        checkNotNull(studyIdentifier);
+        checkArgument(isNotBlank(guid));
         
         SchedulePlan plan = getSchedulePlan(studyIdentifier, guid);
+        
+        forEachCriteria(plan, scheduleCriteria -> deleteCriteria(scheduleCriteria));
         mapper.delete(plan);
     }
+    
+    private void forEachCriteria(SchedulePlan plan, Function<ScheduleCriteria,Criteria> consumer) {
+        if (plan.getStrategy() instanceof CriteriaScheduleStrategy) {
+            CriteriaScheduleStrategy strategy = (CriteriaScheduleStrategy)plan.getStrategy();
+            for (int i=0; i < strategy.getScheduleCriteria().size(); i++) {
+                ScheduleCriteria scheduleCriteria = strategy.getScheduleCriteria().get(i);
+                
+                // Add the key... builder makes this verbose
+                scheduleCriteria = new ScheduleCriteria.Builder()
+                        .withSchedule(scheduleCriteria.getSchedule())
+                        .withCriteria(scheduleCriteria.getCriteria())
+                        .withKey(getKey(plan, i)).build();
+
+                Criteria criteria = consumer.apply(scheduleCriteria);
+
+                // Now add the new criteria object and set new object in list
+                scheduleCriteria = new ScheduleCriteria.Builder()
+                        .withSchedule(scheduleCriteria.getSchedule())
+                        .withCriteria(criteria).build();
+                strategy.getScheduleCriteria().set(i, scheduleCriteria);
+            }
+        }        
+    }
+    
+    private String getKey(SchedulePlan plan, int index) {
+        return "scheduleCriteria:" + plan.getGuid() + ":" + index;
+    }
+    
+    // Save the criteria object if it exists. If it does not, copy the criteria data from the ScheduleCriteria
+    // and save that. We then set that criteria object on the scheduleCriteria. Once all models have a criteria 
+    // object, this will be removed along with the fields on the scheduleCriteria object.
+    private Criteria persistCriteria(ScheduleCriteria scheduleCriteria) {
+        Criteria criteria = scheduleCriteria.getCriteria();
+        Criteria makeCopyOf = (criteria == null) ? scheduleCriteria : criteria;
+
+        Criteria copy = criteriaDao.copyCriteria(scheduleCriteria.getKey(), makeCopyOf);
+        criteriaDao.createOrUpdateCriteria(copy);
+        return copy;
+    }
```

Comment:
createOrUpdateCriteria() already calls copy() internally. After the aforementioned refactor, this could be as simple as a passthrough to createOrUpdateCriteria.